scripts/ci-e2e.sh

```diff
@@ -41,17 +41,21 @@ source "${REPO_ROOT}/hack/util.sh"
 # Verify the required Environment Variables are present.
 capz::util::ensure_azure_envs
 
-export LOCAL_ONLY=${LOCAL_ONLY:-"true"}
```

Comment:
There are a bunch of jobs that currently set this variable and rely on it to not run certain tests in [our job definitions ](https://github.com/jackfrancis/test-infra/blob/c760cd560b643dda6377965fd481f7b2c9aaff30/config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml) (I believe it's part of the anonymous-pull preset)

Merging this PR as-is would break all those jobs. We could update them before merging the PR to set both, or, and maybe that's safer, set the value of `USE_LOCAL_KIND_REGISTRY` to the value of `LOCAL_ONLY` if it's set for backwards compatibility purposes.

Reply:
here where that variable is defined: https://github.com/kubernetes/test-infra/blob/166b008612e2d3319bfead89db593bc3bc52f127/config/prow/config.yaml#L950

Reply:
I think the latter option of maintaining backwards compatibility makes sense, I was just hoping to make the variable name a bit more descriptive.

Reply:
Sure thing, I'll update this in test-infra too. I wasn't planning on backporting it, but do you think it would make sense to do so?

Reply:
not backporting is fine, but you'll have to keep the existing `LOCAL_ONLY` variable for a bit side by side for the jobs that run with the release branches